docs: v0.19.0 release notes (#2985)
spiceai/spiceai: docs/release_notes/v0.19.0-beta.md

Example `spicepod.yml`:

```yaml
datasets:
  - from: abfs://foocontainer/taxi_sample.csv
    name: azure_test
    params:
      azure_account: spiceadls
      azure_access_key: abc123==
      file_format: csv
```

For a full list of supported files, see the [Object Store File Formats](/components/data-connectors/index.md#object-store-file-formats) documentation.

For more details, see the [Azure Blob Storage Data Connector documentation](https://docs.spiceai.org/components/data-connectors/github).

## Breaking Changes

- **Spice.ai Data Connector:** The key for the Spice.ai Cloud Platform Data Connector has changed from `spiceai` to `spice.ai`. To upgrade, change uses of `from: spiceai:` to `from: spice.ai:`.

- **GitHub Data Connector:** Pull Requests column `login` has been renamed to `author`.

- **CLI SQL REPL:** A terminating semi-colon ';' is now required for each executed SQL block.

> - **Spice Runtime hot-reload:** When running the Spice Runtime directly with `spiced`, hot-reload is disabled. Run with `spice run` to enable hot-reload.

Reviewer: ```suggestion
- **Spicepod Hot-Reload:** When running `spiced` directly, hot-reload of spicepod.yml configuration is now disabled. Run with `spice run` to use hot-reload.
```

# Spice v0.19.0-beta (Sep 30, 2024)

The Spice v0.19.0-beta release focuses on fixes and performance for accelerators, improving the coverage of TPC-DS derived queries. Additionally, the release includes some user experience enhancements to the Spice SQL REPL, a new Azure data connector, and some changes to the column formats of some GitHub connector sources. This release also includes fixes and performance improvements for embeddings and models, as well as some dependency updates.

## Highlights in v0.19.0-beta

**SQL REPL UX**: The SQL REPL (`spice sql`) now supports multi-line editing and tab indentation. 

> **GitHub Connectors**: Pull requests column rename `login` to `author`, `labels` changed to a plain string array for both issues and pulls.

Reviewer 2: This should be called out as a breaking change, see previous releases

> **Azure data connector**: New Azure blob storage data connector (`abfss://`)

Reviewer 2: ```suggestion
**Azure data connector**: New Azure blob storage data connector (`abfs://`)
```

## Dependencies

- **[DataFusion Table Providers](<(https://github.com/datafusion-contrib/datafusion-table-providers)>):** Upgraded to rev `826814ab149aad8ee668454c83a0650fb8b18d60`.
- [async-graphql](https://crates.io/crates/async-graphql) updated to 7.0.11
- [async-trait](https://crates.io/crates/async-trait) updated to 0.1.83
- [snafu](https://crates.io/crates/snafu) updated to 0.8.5
- [tempfile](https://crates.io/crates/tempfile) updated to 3.13.0

> - [tonic](https://crates.io/crates/tonic) updated to 0.12.3

Reviewer 2: We don't need to call out these minor dependency upgrades

## Highlights in v0.19.0-beta

**Improved TPC-DS Coverage**: Enhanced support for TPC-DS derived queries.

**CLI SQL REPL**: The CLI SQL REPL (`spice sql`) now supports multi-line editing and tab indentation. Note, a semi-colon ';' is now required for each executed SQL block.

**Azure Storage Data Connector**: A new Azure Blob Storage data connector (`abfs://`) has been added, enabling federated SQL queries on files stored in Azure Blob-compatible endpoints, including Azure BlobFS (`abfss://`) and Azure Data Lake (`adl://`). Supported file formats can be specified using the `file_format` parameter.

Example `spicepod.yml`:

```yaml
datasets:
  - from: abfs://foocontainer/taxi_sample.csv
    name: azure_test
    params:
      azure_account: spiceadls
      azure_access_key: abc123==
      file_format: csv
```

For a full list of supported files, see the [Object Store File Formats](/components/data-connectors/index.md#object-store-file-formats) documentation.

For more details, see the [Azure Blob Storage Data Connector documentation](https://docs.spiceai.org/components/data-connectors/github).

## Breaking Changes

> - **GitHub Data Connector:** Pull Requests column `login` has been renamed to `author`.

Reviewer 3: ### 
```suggestion
- **Spice.ai Data Connector:** The key for the Spice.ai Cloud Platform Data Connector has changed from `spiceai` to `spice.ai`. To upgrade, change uses of `from: spiceai:` to `from: spice.ai:`.

- **GitHub Data Connector:** Pull Requests column `login` has been renamed to `author`.
```
